arnon_clustering_statistics.Rmd

```diff
@@ -0,0 +1,219 @@
+
+# Check if these packages are installed and install them if they aren't
+lapply(packages, function(package) {
+  if (!(package %in% installed.packages())) {
+    install.packages(package)
+  }
+})
+```
+
+Import data that was previously obtained from submitting counts that were formatted
+by `arnon_asap_data_prep.Rmd` to [ASAP](https://asap.epfl.ch/)
+```{r Import data}
+# Get the file names of all the normalized files
+normalized.files <- dir(file.path("arnon_data", "normalized_arnon"))
+
+# Read in each of the normalization files
+normalized.data <- lapply(normalized.files, function(file) {
+                          readr::read_tsv(file.path("arnon_data",
+                                                    "normalized_arnon",
+                                                    file))
+                          })
+
+# Keep all the gene lists
+genes <- lapply(normalized.data, function(x) x[,1])
+
+# Keep the names with it.
+names(normalized.data) <- gsub("\\.tab|arnon_", "", normalized.files)
+```
+
+Import the metadata: 
+```{r Metadata set up}
+# Read in the data
+meta <- readr::read_csv(file.path("arnon_data", "meta_data.csv"))
+```
+
+Retrieve cell type and plate batch for these samples:
+```{r Retrieve cell type and plate batch for these samples}
+# Get sample names from columns
+samples <- colnames(normalized.data[[1]])[-1]
+
+# Keep metadata only for the samples we have
+meta <- meta[match(samples, meta$title), ]
+
+# Cell types info as it's own vector
+cell.types <- as.factor(meta$cell.type.ch1)
+
+# Plate batch info as a factor
+plate.batch <- as.factor(meta$cohort.ch1)
+```
+
+Run tsne clustering for each dataset:
+```{r Make clusters with tsne}
+# Run tsne on each dataset and extract
+tsne <- lapply(normalized.data, function(dataset) {
+               # Run tsne
+               tsne.res <- Rtsne::Rtsne(t(dataset[,-1]), check_duplicates = FALSE)
+               
+               # Only keep the dimension coordinates
+               tsne.res <- tsne.res$Y
+               
+               # Keep the sample names
+               names(tsne.res) <- samples
+              })
+```
+
+Plot all the tsne's for all the datasets with labeling for cell type. 
+```{r Plot the cell type data}
+# Plot with cell type labels
+cell.type.plots <- lapply(tsne, function(dataset) {
+                          # Get data normalizaion name
+                          set.name <- names(tsne)[parent.frame()$i[]]
+                          
+                          # Make plots for cell type and plate batch 
+                          DimPlot(dataset, cell.types, xlabel = "tsne dim 1",
+                                  ylabel = "tsne dim 2", name = set.name)
+                    })
+
+# Extract the legend
+legend <- cowplot::get_legend(cell.type.plots[[1]])
+
+# Surpressing the legend in the files
+cell.type.plots <- lapply(cell.type.plots, function(a.plot) {
+                          a.plot + theme(legend.position = 'none')
```

Comment:
Can you move this bit to above (lines 92-93)

```R
DimPlot(dataset, cell.types, xlabel = "tsne dim 1",  
        ylabel = "tsne dim 2", name = set.name) +
  theme(legend.position = 'none')
```

And snag the legend some other way?